### PR TITLE
Some fixes

### PR DESCRIPTION
**Pull request by ramiresviana**
### Directory conflict checking

- File scan is using path attribute for uploaded directory items while the conflict checking method uses fullPath item attribute, causing conflict checking issues when uploading folders. The directory items returned by scan method should have fullPath attribute instead of path.

### Dark theme missing colors #1022 #1025

- Share box, share prompt and listing error message are missing dark theme colors. Input validation colors are missing from user profile password field. The surface colors are kinda out of the material design guidelines. Added new colors following the dark theme material guidelines, with branded dark surfaces. Added dark styles for share box, share prompt, listing error message and other small elements.

### Prompt when closing windows with active uploads

- The post frontend file api method sets the window beforeunload event for every new upload, and when the first upload is finished the event listener is removed, causing no confirmation prompt for the next pending uploads. The beforeunload listener should be set when the first upload is started and should be removed when the last upload was finished.

### Remove incomplete uploaded files

- When a upload is being made, there are a serie of issues that the frontend can go through while the file is being transferred, like losing the connection or a browser crash. The incomplete uploaded file is available on the server, without the user being aware that it is a corrupt file. Upload requests that ends prematurely or in unexpected way should have its file removed from the disk.

### Blinking previewer #976

- Navigating between files in previewer mode cause the preview component destroyed and its visibility toggled. A state is required to keep the component rendered between navigation.

### CODE REVIEW

**Comment by o1egl** [comment]
Thank you for your contribution. I found that preview works better but it still blinking. 
Check out this recoding:
[IMAGE ALT TEXT HERE](https://www.youtube.com/watch?v=xdjMIaQ5Bgo)

**Comment by ramiresviana** [comment]
@o1egl i've reviewed my approach to keep components rendered all time to prevent blinking.

- Added state for the file list in the previewer, instead of making a new request every time the previewer changes.
- Buttons will be rendered all times, but will remain in a disabled state during loading.
- The previewer will show images in the center of the screen while loading
- Resizing the screen will now cause the position of previewer images to be recalculated based on the center of the screen.

**Comment by ramiresviana** [comment]
### Check file conflict when copying or moving

- Using copy or move buttons can cause conflict errors when the destination have files with the same name. Copying, moving and pasting actions should have conflict checking to give the user options for replacing the files.

### Option for renaming files with conflict

- Is not possible to keep both files in a conflict scenario, since the only option available is to overwrite existing files. The rename option will add a prefix to conflicting files, allowing the user to have both files. When a file with the prefix already exists, the prefix will be appended until a file with the new name dont exists.

### Duplicate files in the same directory #1011 

- With the renaming conflict files feature, the block of duplicating files in the same directory can be removed. Duplicate files in the same directory will be renamed automatically without prompt.

### Reset clipboard after pasting cutted files

- Keeping cutted files in the clipboard state after pasting then, will cause not found error in the next pasting, because the file location was changed. Clipboard should reset after pasting cutted files.

File: http/resource.go [hunk 1]
```diff
@@ -144,6 +148,27 @@ var resourcePatchHandler = withUser(func(w http.ResponseWriter, r *http.Request,
...
+	rename := r.URL.Query().Get("rename") == "true"
+
+	if !override && !rename {
+		if _, err = d.user.Fs.Stat(dst); err == nil {
+			return http.StatusConflict, nil
+		}
+	}
+
+	if rename {
+		suffix := "_copy"
```
Comment (o1egl) [inline http/resource.go]: It might be a file with a different content. Instead of a copy let's add (N) suffix

File: http/resource.go [hunk 2]
```diff
@@ -144,6 +148,27 @@ var resourcePatchHandler = withUser(func(w http.ResponseWriter, r *http.Request,
...
+		_, err = d.user.Fs.Stat(dst)
+		exists := err == nil
+
+		for exists {
+			dst = dst + suffix
```
Comment (o1egl) [inline http/resource.go]: suffix should be put before file extension.

File: http/resource.go [hunk 3]
```diff
@@ -144,6 +148,27 @@ var resourcePatchHandler = withUser(func(w http.ResponseWriter, r *http.Request,
...
+			return http.StatusConflict, nil
+		}
+	}
+
+	if rename {
+		suffix := "_copy"
+		_, err = d.user.Fs.Stat(dst)
+		exists := err == nil
+
+		for exists {
```
Comment (o1egl) [inline http/resource.go]: I think better would be to rewrite it like:
```go
for {
	// ...
	if notExist {
		break;
	}
}
```
With that you can reduce code duplications

File: http/resource.go [hunk 4]
```diff
@@ -144,6 +148,27 @@ var resourcePatchHandler = withUser(func(w http.ResponseWriter, r *http.Request,
 		return http.StatusForbidden, nil
 	}
 
+	override := r.URL.Query().Get("override") == "true"
+	rename := r.URL.Query().Get("rename") == "true"
+
+	if !override && !rename {
+		if _, err = d.user.Fs.Stat(dst); err == nil {
+			return http.StatusConflict, nil
```
Comment (o1egl) [inline http/resource.go:156]: Error must be triggered only if src != dst

**Review by o1egl — CHANGES_REQUESTED** [review]

**Comment by o1egl** [comment]
@ramiresviana can you please squash blinking commits into a single commit

File: http/resource.go (hunk 4 above)
Comment (ramiresviana) [reply http/resource.go:156]: This verification is required when duplicating files in the same folder, in a conflict scenario the frontend need to inform that the files should be renamed.

**Comment by ramiresviana** [comment]
@o1egl rename suffix is now in the (N) incremental format.

https://github.com/filebrowser/filebrowser/blob/2636f876ab8f88eea6d9548de524ca2339eb0843/http/resource.go#L160-L174

**Review by o1egl — APPROVED** [review]

**Pull request merged**